### PR TITLE
fix(condo): DOMA-8031 fixed a very loooong news loading

### PR DESCRIPTION
**Pull request by Alllex202**
Problem: `NewsItem` can has many (may be 100+) scopes which are loaded on the table page in CRM .it may be more than 10 seconds
Resolve: add `compactScopes` virtual field. It prepares data for displaying addresses in CRM without fetching all scopes

<details>
	<summary>BEFORE</summary>

https://github.com/open-condo-software/condo/assets/56914444/1caba5ca-a0b4-457c-ba96-91247fc54a59
</details>
<details>
	<summary>AFTER</summary>

https://github.com/open-condo-software/condo/assets/56914444/12b71524-39a6-40d4-b879-b5b1a24a1d86
</details>

### CODE REVIEW

File: packages/keystone/schema.js [hunk 1]
```diff
@@ -143,6 +143,31 @@ async function find (schemaName, condition) {
...
+ * @param {string[]?} args.sortBy
+ * same as "orderBy", only in an array
+ * @param {string?} args.search
+ * "search" not supported in knex adapter (from knex adapter implementation)
+ * @param {boolean} meta
+ * If "true", then returns { count } with the number of objects.
+ * If "false", it will return objects
+ * @param from
+ */
+async function itemsQuery (schemaName, args, { meta = false, from = {} } = {}) {
```
Comment (AleX83Xpert) [inline packages/keystone/schema.js:164]: Is it related to current PR?

File: apps/condo/domains/news/utils/clientSchema/NewsRenders.tsx [hunk 2]
```diff
@@ -111,13 +108,13 @@ export const ResendNewsButton = ({ intl, newsItem }) => {
 
 export const getTypeRender: GetTypeRenderType = (intl, search) => (text, newsItem) => {
     const CommonTypeMessage = intl.formatMessage({ id: 'news.type.common' })
-    const ЕmergencyCommonTypeMessage = intl.formatMessage({ id: 'news.type.emergency' })
+    const EmergencyCommonTypeMessage = intl.formatMessage({ id: 'news.type.emergency' })
```
Comment (AleX83Xpert) [inline apps/condo/domains/news/utils/clientSchema/NewsRenders.tsx:111]: :)

File: apps/condo/domains/news/schema/NewsItem.js [hunk 3]
```diff
@@ -153,6 +153,40 @@ const NewsItem = new GQLListSchema('NewsItem', {
             },
         },
 
+        compactScopes: {
+            schemaDoc: 'Returns the number of scopes that are specified for sending the news, and also the first two of them.\n' +
+                'Used to reduce requests for get of scopes in the UI',
+            type: 'Virtual',
+            extendGraphQLTypes: ['type ShortScopesField { count: Int!, firstTwo: [NewsItemScope]! }'],
```
Comment (AleX83Xpert) [inline apps/condo/domains/news/schema/NewsItem.js:160]: Maybe not use array size as a part of the name of the variable?
```suggestion
            extendGraphQLTypes: ['type ShortScopesField { count: Int!, firstOnes: [NewsItemScope]! }'],
```

File: apps/condo/domains/news/schema/NewsItem.js [hunk 4]
```diff
@@ -153,6 +153,40 @@ const NewsItem = new GQLListSchema('NewsItem', {
...
+            graphQLReturnType: 'ShortScopesField',
+            resolver: async (item) => {
+                const firstTreeScopes = await itemsQuery('NewsItemScope', {
+                    where: {
+                        newsItem: { id: item.id }, deletedAt: null,
+                    },
+                    first: 3,
```
Comment (AleX83Xpert) [inline apps/condo/domains/news/schema/NewsItem.js:167]: Maybe move the number of scopes in some constant.
```suggestion
                    first: SOME_CONSTANT + 1,
```

File: packages/keystone/schema.js (hunk 1 above)
Comment (Alllex202) [reply packages/keystone/schema.js:164]: Yes, we need this function in this pr

File: apps/condo/domains/news/utils/clientSchema/NewsRenders.tsx (hunk 2 above)
Comment (Alllex202) [reply apps/condo/domains/news/utils/clientSchema/NewsRenders.tsx:111]: There were letters from different languages)

**Review by AleX83Xpert — APPROVED** [review]

**Pull request merged**